### ISSUE TITLE
After running logout() cannot add new data to session

### ISSUE THREAD

**Issue by Pizzicato** [benedmunds/CodeIgniter-Ion-Auth#1146]
Hi! Thanks for developing this library, it's really useful! :-D
I'm having a problem when logging out a user and adding data to the session afterwards. Once redirected the session is empty. This is my controller logout code:

```
    public function delete($url = '')
    {
        $url = $url ? base64_url_decode($url) : site_url_lang('admin/login');
        if($this->ion_auth->logged_in()){
            $this->ion_auth->logout();
            $_SESSION['status'] = $this->ion_auth->messages();
            $this->session->mark_as_flash('status');
        }

	    redirect($url);
    }
```
$_SESSION before the redirect is:
`array(7) { ["__ci_last_regenerate"]=> int(1509623879) ["identity"]=> string(13) "administrator" ["email"]=> string(15) "admin@admin.com" ["old_last_login"]=> int(1509623311) ["last_check"]=> int(1509623871) ["status"]=> string(23) "Logged Out Successfully" ["__ci_vars"]=> array(1) { ["status"]=> string(3) "new" } } `
But after the redirect:
`array(1) { ["__ci_last_regenerate"]=> int(1509623957) } `

I'm using the last current codeigniter version (3.16). My server runs PHP 5.4

Cheers

**Comment by benedmunds** [comment]
I *think* that CodeIgniter might not let you use the new session until after the redirect.  I'm not sure but it used to be really picky about that.

Might be worth just setting a URL param for this and checking for that, something like:


`public function delete($url = '')
    {
        $url = $url ? base64_url_decode($url) : site_url_lang('admin/login');
        if($this->ion_auth->logged_in()){
            $this->ion_auth->logout();
            $url .= '?logged_out=1'
        }

	    redirect($url);
    }`

Or you could troubleshoot the session destroy/create/regenerate logic here: https://github.com/benedmunds/CodeIgniter-Ion-Auth/blob/2/libraries/Ion_auth.php#L410

It has logic for different CI/PHP versions so just note which piece of this logic you're hitting in your environment.

**Comment by Pizzicato** [comment]
Hi,
Thank you for answering so fast!
I checked the code in the ion_auth.php file.

I don't understand why `session_start()` is called only if the server PHP version is above 7.0. Actually, if I remove the condition block and `session_start()` gets called it works. Maybe it's a bug...

Cheers

**Comment by benedmunds** [comment]
It was originally introduced to fix a PHP7 issue: https://github.com/benedmunds/CodeIgniter-Ion-Auth/issues/906

What if you use the CI Session lib instead of using PHP session directly?

**Comment by Pizzicato** [comment]
It makes no difference if I use CI Session lib, actually in the CodeIgniter user guide they recommend using the $_SESSION array: [here](https://www.codeigniter.com/user_guide/libraries/sessions.html#CI_Session::set_flashdata)

I've changed my code to this:
```
    public function delete($url = '')
    {
        $url = $url ? base64_url_decode($url) : site_url_lang('admin/login');
        if($this->ion_auth->logged_in()){
            $this->ion_auth->logout();
            session_start();
            $this->session->sess_regenerate(TRUE);
            $_SESSION['status'] = $this->ion_auth->messages();
            $this->session->mark_as_flash('status');
        }

	    redirect($url);
    }
```
This way it works, but obviously it's not the ideal way to do it.

I'm not sure if the fix for #906 is correct for some PHP 5 versions, like mine.

Cheers

**Comment by benedmunds** [comment]
Hmm yea maybe.  Good hack around it there at least.

If anyone else reports issues I'll create a few environments and see if just removing that PHP version check causes issues on any other versions.

Thanks!
